src/core/contracts/utilities.ts

```diff
@@ -123,16 +124,35 @@ export const subscribeToEvent = async (
     }
   });
 
+  postBlockNumber = await provider.getBlockNumber();
+
   const pastLogs = await provider.getLogs({
     ...filter,
     fromBlock: fromBlock,
-    toBlock: currentBlockNumber,
+    toBlock: postBlockNumber,
   });
 
-  [...pastLogs, ...newLogsQueue].forEach((log) => {
-    doReceiveEvent(log);
-  });
+  // Overlap is really unlikely, but possible.
```

Comment:
it took me a minute to understand what was happening here. Not sure if a comment explaining what we are trying to achieve would be helpful?